src/database/migrations/0040-create-sync_basefields.sql

```diff
@@ -0,0 +1,20 @@
+CREATE TYPE sync_basefields_status AS ENUM (
+  'pending',
+  'in_progress',
+  'completed',
+  'failed',
+  'canceled'
+);
+
+CREATE TABLE sync_basefields (
```

Comment:
@jasonaowen @slifty I like the idea of using a nested tasks endpoint. It would nicely follow the creation of an abstract 'task' interface which bulk_upload and basefields_copy could inherit. Would we want that refactor for bulk_uploads to be part of this pr? It would likely extend it by quite a bit. @slifty interested to hear your thoughts on that, either way I think basefields_copy or basefields_copy_*task* is a much better name for the entity

Reply:
I agree with everything @jasonaowen said!

@hminsky2002 I would include that as work to be accomplished before merging this PR -- otherwise this work will be somewhat refactored and the refactor PR will be just as large as this one would be, but with the negative of churn.

You might consider making that work a separate PR, though (to refactor bulk uploads to `/tasks`) and then rebase this PR on top of that.

Reply:
Just to reference here, resolved by PR #1305